python/cudf/cudf/core/series.py

```diff
@@ -1419,7 +1419,7 @@ def as_mask(self):
         -------
         device array
         """
-        return cudautils.compact_mask_bytes(self.to_gpu_array())
+        return cudautils.compact_mask_bytes(self.mask.mem)
```

Comment:
You want the data here as you're compressing a column of booleans
```suggestion
        return cudautils.compact_mask_bytes(self.data.mem)
```